components/timezone/src/ixdtf.rs

```diff
@@ -227,17 +254,26 @@ impl<'a> Intermediate<'a> {
         };
         Ok(Self {
             offset,
+            is_z,
             iana_identifier,
             date,
             time,
         })
     }
 
     fn offset_only(self) -> Result<UtcOffset, ParseError> {
-        let Some(offset) = self.offset else {
+        let None = self.iana_identifier else {
             return Err(ParseError::MismatchedTimeZoneFields);
         };
-        let None = self.iana_identifier else {
+        if self.is_z {
+            if let Some(offset) = self.offset {
+                if offset != UtcOffsetRecord::zero() {
+                    return Err(ParseError::RequiresCalculation);
```

Comment:
This is rejecting `Z[-0800]`, right? Seems like `RequiresCalculation` isn't quite the right error case?

Reply:
We need to subtract 8h from the timestamp, which is definitely a calculation. The only non-calculation offset is `Z[+0000]`.